src/cargo/util/toml.rs

```diff
@@ -109,9 +109,7 @@ pub fn to_manifest(contents: &str,
     let manifest: TomlManifest = serde_ignored::deserialize(root, |path| {
         let mut key = String::new();
         stringify(&mut key, &path);
-        if !key.starts_with("package.metadata") {
```

Comment:
Hm, I think this if is intentional to avoid warning about "key metadata is unused". Do we want to start issuing this warning?